dvc/cli.py

```diff
@@ -103,11 +140,48 @@ def _find_parser(parser, cmd_cls):
 class DvcParser(argparse.ArgumentParser):
     """Custom parser class for dvc CLI."""
 
+    cmd_choices = {}
+    hidden_cmds = []
+
     def error(self, message, cmd_cls=None):  # pylint: disable=arguments-differ
         logger.error(message)
         _find_parser(self, cmd_cls)
 
     def parse_args(self, args=None, namespace=None):
+        # NOTE: this is a custom check to see if any suggestions can
+        # be displayed to users in case a command contains typos
+        # E.g. `dvc commti` would display
+        # The most similar command is
+        #         commit
+        if args is None:
+            args = sys.argv[1:]
+        else:
+            args = list(args)
+        if (
+            len(args) >= 1
+            and args[0] not in self.cmd_choices
+            and args[0] not in self.hidden_cmds
```

Comment:
Could we make these private properties (e.g. cmd_choices -> _cmd_choices) and get them dynamically instead of calling set_cmd_choices() explicitly?